src/transformers/models/bart/modeling_bart.py

```diff
@@ -237,10 +238,7 @@ def forward(
             )
 
         if attention_mask is not None:
-            if attention_mask.size() != (bsz, 1, tgt_len, src_len):
-                raise ValueError(
-                    f"Attention mask should be of size {(bsz, 1, tgt_len, src_len)}, but is {attention_mask.size()}"
-                )
+            attention_mask = attention_mask[:, :, :, : key_states.shape[-2]]
```

Comment:
If we remove the `+1` in L801 (`else past_seen_tokens + sequence_length + 1`) I think we can remove this line (and similar slicing in SDPA)